Design.md

```diff
@@ -136,46 +121,88 @@ This API will update the existing storage tenant with the provided `cd_tenant_id
 
 1. Vault `update-account` api will be called using vaultclient with the provided `tenantID` as `AccountID` and `tenant` properties
 
+### `SetupAssumeRole` subroutine
+1. Use vaultclient to call `generate-account-access-key` with `expirationTime` for the account.
+1. Create a role using the generated access key with:
+    1. Role name: `osis` (So role-arn can be generated in the format `arn:aws:iam::[account-id]:role/osis`)
+    1. Trust policy:
+   ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [{
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "service-name"
+            },
+            "Action": "sts:AssumeRole"
```

Comment:
Is `AssumeRole` the only action needed after we assumeRole?
Should it be `iam:*`?

Reply:
Here its mentioned with full s3 and I am access: https://github.com/scality/vmware-ose-scality/pull/23/files#diff-eecbe2ff4990ae72ef9831ff0ab35dc789658a802cc6aae8c854534b68202f3aR157

Reply:
This is a trust inline policy attached to the role while creating to allow role to assumeRole. This policy enables superadmin to use AssumeRoleBackbeat using this role.

"Full S3 & IAM access" is a managed policy that will be created and attached to the role in the later steps